### ISSUE TITLE
Summon KeepassXC to current desktop

### ISSUE THREAD

**Issue by bluppfisk** [keepassxreboot/keepassxc#1858]
In a multi-desktop environment like Ubuntu, when toggling the KeepassXC window (e.g. through the menu in its tray icon), the window is opened on the desktop where it was last put. As KeepassXC is a utility that should require as little navigating as possible, it would be much preferable if KeepassXC always opened on the current desktop.

## Debug Info
KeePassXC - Version 2.3.1
Revision: 2fcaeea

Libraries:
- Qt 5.5.1
- libgcrypt 1.8.1

Operating system: Ubuntu 16.04.4 LTS
CPU architecture: x86_64
Kernel: linux 4.15.7-041507-generic

Enabled extensions:
- Auto-Type
- Browser Integration
- Legacy Browser Integration (KeePassHTTP)
- SSH Agent
- YubiKey

**Comment by droidmonkey** [comment]
Is that possible? Seems like this is handled by the window manager (and user choice) not the application.

**Comment by TheZ3ro** [comment]
I don't think this is handled by KeePassXC or Qt, I think this is handled by the desktop environment/window manager

**Comment by bluppfisk** [comment]
Perhaps, but there are plenty of apps that reside in the tray and create their windows on whatever desktop you're currently on. Example would be Dropbox' preferences window. Maybe there's a difference between modals and main windows, but it's worth investigating. I believe that even the 'old' Keepass on Mono always opens on the right desktop.

**Comment by phoerious** [comment]
My X11 knowledge is limited, but there are various window flags in the freedesktop standard which target virtual desktops.

**Comment by frederickjh** [comment]
With KeePassX It was possible to right-click the top bar of the window and select **Always on Visible WorkSpace**.  However with KeePassX it never disappeared totally to the taskbar.

This works on KeePassXC until you toggle the window then this setting is lost. If this setting could persist then this would solve this issue as one can then set the window to appear on all workspaces.

The other option that the user can use now is to uncheck **Hide window to system tray when minimized** and set the window to appear on all desktop, per instructions above.

**Comment by rpedrica** [comment]
Confirming the issue here with KDE5/Plasma and multiple desktops. I have 2 physical screens with 2 desktops. If Keepassxc was last opened on desktop B and I'm currently on desktop A, on invoking global shortcut/auto-type in a browser tab, keepassxc shows focus in desktop B (ie. you can go to desktop B and Keepassxc brings up the selection for which account/pwd combo to use) - even after selecting the correct account/pwd details to insert, this is not inserted into the account/pwd field in the browser on desktop A.

To get this to work, I need to move Keepasscx to desktop A (right-click title bar and Move to Desktop -> A) and then do shortcut again for auto-type to work.

It works fine when there is only 1 possible result for a site, in which case there is no selection box and the auto-type works as expected. The problem exists when there is more than 1 result for a site ie. multiple logins on the same site.

**Comment by frederickjh** [comment]
On Ubuntu 16.04 with Unity I am finding that, if I have the KeePassXC window set to  **Always on Visible WorkSpace** then all I need to do is us the taskbar **Toogle Window** button. Sometimes I have to do it twice. Put it always appears.

**Comment by am-oxipay** [comment]
This  would also be super valuable.  I think it sort of works currently but only if it's not being displayed on a desktop already. 

i.e if I open keepassxc on desktop one, then navigate to desktop tw and toggle the system tray icon, it won't move the window to the current screen.

**Comment by masenocturnal** [comment]
I have this challenge. I think that what happens is that KeepassXC is visible on another virtual desktop (i.e not the one you're on). 

You press the tray icon (in plasma) expecting it to come to you (like all the other applications that are in your system tray), however it doesn't. 

If the window is minimised to the system tray already then the application behaves "as expected" - at least by Plasma users. 

I would be surprised if Signal, Nehko, Element, slack etc.. all have custom rules specifically for plasma and I don't appear to have any custom Kwin rules for any of these applications . 

I don't really know how to work out what is different about keepassxc to other applications but with some guidance I would be happy to debug further .

**Comment by Chealer** [comment]
@bluppfisk wrote:
> In a multi-desktop environment like Ubuntu, when toggling the KeepassXC window (e.g. through the menu in its tray icon), the window is opened on the desktop where it was last put.

I would not call _Ubuntu_ “a multi-desktop environment”; a Ubuntu session's support for multiple desktops depends on the desktop environment managing that session.

As this is an ITS, each ticket should emphasize the problem, in particular in the title. Phrased as it currently is, your report was interpreted as a request for enhancement, but I understand from your ticket's body that the problem is KeePassXC opens on another desktop, which―to me―would just be a bug (unless―perhaps―you specially configured that behavior). On Microsoft Windows, there are programs which open on a different desktop when they are _restored_ (for example following a session restart), but I cannot think of any program which _initially_ opens on a different desktop.

Please check if this reproduces with a recent version, and if so, indicate the version(s) you reproduce with, provide precise reproduction steps (probably very few in this case), and specify the precise compositing manager you use.
For example, if this occurs with GNOME (Ubuntu‘s default desktop environment), you could retitle to something like:
> Sometimes opens on a different virtual desktop (workspace) on GNOME

There are several terms to designate what you call a “desktop”. The most precise term seems to be _[virtual desktop](https://en.wikipedia.org/wiki/Virtual_desktop)_, but Windows simply calls it “desktop”, and most FLOSS compositing managers call that a “workspace”.

@droidmonkey wrote:
> Is that possible?

It's certainly possible in general; KeePassXC 2.7.12 opens on the current virtual desktop on my Windows 11 install. It could be that it takes more work on some desktop environments, but I have never seen that on the very few I am familiar with.

[🅭🄍](https://www.philippecloutier.com/Common+infrastructure+licensing#its)

**Comment by LinAGKar** [comment]
@masenocturnal:

>I have this challenge. I think that what happens is that KeepassXC is visible on another virtual desktop (i.e not the one you're on).

>You press the tray icon (in plasma) expecting it to come to you (like all the other applications that are in your system tray), however it doesn't.

It doesn't even have to involve multiple desktops. From my experience (on OpenSUSE Tumbleweed with Plasma Wayland), what happens when clicking the tray icon is:

- If KeepassXC has no window open: The window opens, in the foreground on the current desktop
- If KeepassXC has a window open and it's currently in focus: The window is closed
- If KeepassXC has a window open but it's not currently in focus: Nothing happens

The third case could be when it's on another desktop, when it's behind another window, or any other case when the KeepassXC window is not the one in focus.

In the third case I would expect clicking it would either close the window, or bring it to the foreground. Clicking "show/hide window" in the right-click menu in the third case does close the window (in which case a second click on the same thing then opens it on the current desktop), and clicking the tray icon for many other applications seems to do the same.
